### ISSUE TITLE
Support prediction on TensorFlow's saved model

### ISSUE THREAD

**Issue by Tomczik76** [spotify/scio#1043]
I'm testing using the ``` "com.spotify" %% "scio-tensorflow" % "0.4.7"``` package and I'm having issues with reading my Saved Model with the default way passing a binary file path to ```.predict``` 

The way I'm writing the binary file is 

```
val session = SavedModelBundle.load(graphUri, "serve")
Files.write(Paths.get("./model.bin"), session.graph().toGraphDef)
session.close()
```

The problem is using this method I'm getting errors like 

```
Caused by: java.lang.IllegalStateException: Attempting to use uninitialized value rnn/basic_lstm_cell/weights
[[Node: rnn/basic_lstm_cell/weights/read = Identity[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"](rnn/basic_lstm_cell/weights)]]
	at org.tensorflow.Session.run(Native Method)
	at org.tensorflow.Session.access$100(Session.java:48)
	at org.tensorflow.Session$Runner.runHelper(Session.java:295)
	at org.tensorflow.Session$Runner.run(Session.java:245)
```

Although I've got it working by creating my own custom ```DoFn``` based on the ```PredictDoFn```, only instead of passing the ```DistCache[Array[Byte]]``` I pass in the Uri to my Saved Model and create the Tensorflow session like

```s = SavedModelBundle.load(graphUri, "serve").session()```

Using this method I'm not receiving the error above. I assume that the TensorFlow variables aren't stored in the Graph itself.

Am I making a mistake how writing my Graph to a file?

**Comment by richwhitjr** [comment]
The TF Java api is a little odd compared to most of other TF in that it expects the graph to be frozen.  This means the variables have to be stored as constants in the protobuf file instead in a separate variables directory as TF normally does.

This PR shows how you can freeze the graph:  https://github.com/spotify/spotify-tensorflow/pull/26

**Comment by Tomczik76** [comment]
Do you think it would be worthwhile to create a PR with an implementation of ``` .predict``` that can take a path a standard Saved Model and download it with the FileSystems API and load the session the way I described above?

**Comment by ravwojdyla** [comment]
@Tomczik76 sure. Current `predict` expects frozen graph, which you can simply do by using helper from https://github.com/spotify/spotify-tensorflow.

That said I'm +1 to add `predict` with support for `SavedModelBundle`.

Will go ahead and rename this issue to support `SavedModelBundle` in `predict`.

**Comment by ravwojdyla** [comment]
Fixed in #1162
